process/message.go

```diff
@@ -303,6 +309,56 @@ func (precommit *Precommit) String() string {
 	return fmt.Sprintf("Precommit(Height=%v,Round=%v,BlockHash=%v)", precommit.Height(), precommit.Round(), precommit.BlockHash())
 }
 
+// Resyncs is a wrapper around the `[]Resync` type.
+type Resyncs []Resync
+
+// Resync previous messages.
+type Resync struct {
+	signatory id.Signatory
+	sig       id.Signature
+	height    block.Height
+	round     block.Round
+}
+
+func NewResync(height block.Height, round block.Round) *Resync {
+	return &Resync{
+		height: height,
+		round:  round,
+	}
+}
+
+func (resync *Resync) Signatory() id.Signatory {
+	return resync.signatory
+}
+
+func (resync *Resync) SigHash() id.Hash {
+	return sha256.Sum256([]byte(resync.String()))
+}
+
+func (resync *Resync) Sig() id.Signature {
+	return resync.sig
+}
+
+func (resync *Resync) Height() block.Height {
+	return resync.height
+}
+
+func (resync *Resync) Round() block.Round {
+	return resync.round
+}
+
+func (resync *Resync) BlockHash() id.Hash {
+	return id.Hash{}
+}
```

Comment:
It might be better if this function panics, because it probably should never be called. As it is, if someone accidentally calls this function it would "fail silently" and the caller would proceed with the empty block hash. This would bubble the error elsewhere and make it hard to track down, or in the worst case could cause inconsistent/corrupted state. Panicking would at least be immediately noticeable and traceable.

An alternative: the block hash should be equal to the requested height/round given in the other message fields. This would require that height/round is interpreted as the last known height/round (which would make sense, given the context of a resync message).